cmd/lotus-storage-miner/sectors.go

```diff
@@ -440,69 +444,155 @@ var sectorsExtendCmd = &cli.Command{
 		defer nCloser()
 
 		ctx := lcli.ReqContext(cctx)
-		if !cctx.Args().Present() {
-			return xerrors.Errorf("must pass at least one sector number")
-		}
 
-		maddr, err := nodeApi.ActorAddress(ctx)
+		maddr, err := getActorAddress(ctx, cctx)
 		if err != nil {
-			return xerrors.Errorf("getting miner actor address: %w", err)
+			return err
 		}
 
-		sectors := map[miner.SectorLocation][]uint64{}
+		var params []miner3.ExtendSectorExpirationParams
 
-		for i, s := range cctx.Args().Slice() {
-			id, err := strconv.ParseUint(s, 10, 64)
-			if err != nil {
-				return xerrors.Errorf("could not parse sector %d: %w", i, err)
+		if cctx.Bool("v1-sectors") {
+			extensions := map[miner.SectorLocation]map[abi.ChainEpoch][]uint64{}
+			// are given durations within a week?
+			closeEnough := func(a, b abi.ChainEpoch) bool {
+				diff := a - b
+				if diff < 0 {
+					diff = b - a
+				}
+
+				return diff <= 7*builtin.EpochsInDay
 			}
 
-			p, err := api.StateSectorPartition(ctx, maddr, abi.SectorNumber(id), types.EmptyTSK)
+			sis, err := api.StateMinerActiveSectors(ctx, maddr, types.EmptyTSK)
 			if err != nil {
-				return xerrors.Errorf("getting sector location for sector %d: %w", id, err)
+				return xerrors.Errorf("getting miner sector infos: %w", err)
+			}
+
+			for _, si := range sis {
+				if si.SealProof < abi.RegisteredSealProof_StackedDrg2KiBV1_1 {
+
+					ml := builtin3.SealProofPoliciesV11[si.SealProof].SectorMaxLifetime
```

Comment:
Let's move this into `chain/actors/policy` and make it a function over the version.